x-pack/packages/ml/aiops_components/src/dual_brush/dual_brush.tsx

```diff
@@ -166,23 +155,54 @@ export function DualBrush({
 
             newWindowParameters.baselineMin = px2ts(newBaselineMin);
             newWindowParameters.baselineMax = px2ts(newBaselineMax);
-            newBrushPx.baselineMin = newBaselineMin;
-            newBrushPx.baselineMax = newBaselineMax;
+          }
+
+          const snappedWindowParameters = snapTimestamps
```

Comment:
I find this 200 LOC effect quite difficult to follow. Also, it creates a bunch of callbacks. Did you consider splitting it up?

Reply:
I agree this is complex code, but that's already due to the original implementation and the "trickyness" of the `d3-brush` API to get this work. This update fixes a bug where on a resize and the change of the provided width the transformation from pixels to timestamps would return different values where they should stay the same. At this stage after FF IMHO we should avoid a larger rewrite. Since I wrote the original implementation I can add some comments to explain what the code is doing in a follow up. Once we have better test coverage we could think of some refactor in the next release.

Reply:
I missed the fact it's for 8.4. Indeed we should avoid refactoring after the FF, but worth revisiting later on for 8.5 👍🏻